core/trino-main/src/main/java/io/trino/connector/informationschema/InformationSchemaPageSource.java

```diff
@@ -105,6 +107,10 @@ public InformationSchemaPageSource(
         table = tableHandle.getTable();
         prefixIterator = Suppliers.memoize(() -> {
             Set<QualifiedTablePrefix> prefixes = tableHandle.getPrefixes();
+            if (table == InformationSchemaTable.COLUMNS && prefixes.equals(defaultPrefixes("hive"))) {
+                throw new TrinoException(NOT_SUPPORTED, "Queries which scan all table columns in Hive are expensive " +
+                        "and will eventually fail, so they are not supported. See PRESTO-1822 for more info");
```

Comment:
That's not necessarily true. It depends on how many tables/columns exist in the Hive metastore.